lib/net.c

```diff
@@ -1641,42 +1629,38 @@ init_net_stage1(void *arg)
 
 	calculate_net_config_para(net_conf);
 
-	if (net_conf->gatekeeper_pktmbuf_pool == NULL) {
-		net_conf->gatekeeper_pktmbuf_pool =
-			rte_calloc("mbuf_pool", net_conf->numa_nodes,
-				sizeof(struct rte_mempool *), 0);
-		if (net_conf->gatekeeper_pktmbuf_pool == NULL) {
-			G_LOG(ERR, "net: %s: out of memory\n", __func__);
-			return -1;
-		}
-	}
-
-	/* Initialize pktmbuf pool on each used NUMA node. */
-	for (i = 0; i < net_conf->numa_nodes; i++) {
+	/* Initialize pktmbuf pool on each used lcore. */
+	for (i = 0; i < RTE_MAX_LCORE; i++) {
 		char pool_name[64];
 		int ret;
 
-		if (!net_conf->numa_used[i] ||
+		if (!net_conf->lcores_used[i] ||
 				net_conf->gatekeeper_pktmbuf_pool[i] != NULL)
 			continue;
 
 		ret = snprintf(pool_name, sizeof(pool_name), "pktmbuf_pool_%u",
 			i);
 		RTE_VERIFY(ret > 0 && ret < (int)sizeof(pool_name));
 		net_conf->gatekeeper_pktmbuf_pool[i] =
-			rte_pktmbuf_pool_create(pool_name,
+			rte_mempool_create(pool_name,
```

Comment:
Why do we need to create memory pools directly?

Reply:
``rte_pktmbuf_pool_create()`` doesn't allow us to create memory pools specifying the single-consumer mode.

Reply:
What about `rte_pktmbuf_pool_create_by_ops()`?

Reply:
The above function will call [rte_mempool_create_empty()](https://github.com/cjdoucette/dpdk/blob/fed4cdb1be34ea0016cc8956af2baa40cb67fc7d/lib/librte_mbuf/rte_mbuf.c#L132-L133), specifying the last parameter as ``0``, i.e., multi-consumer, multi-producer.

Reply:
`rte_pktmbuf_pool_create_by_ops()` calls `rte_mempool_create_empty()` passing zero as the last parameter, so it can have a better error message while setting the operations with `rte_mempool_set_ops_byname()`. You find the code of `rte_pktmbuf_pool_create_by_ops()` in `dpdk/lib/librte_mbuf/rte_mbuf.c`.

Reply:
Should we have a separate DPDK function similar to ``rte_pktmbuf_pool_create_by_ops()`` which allows to specify the work mode (i.e., single-consumer)?

Reply:
No, just pass the appropriate parameter to `rte_pktmbuf_pool_create_by_ops()`. You may need to read DPDK's code to find the string we need.

Reply:
Got it! Indeed, [rte_mempool_create()](https://github.com/cjdoucette/dpdk/blob/fed4cdb1be34ea0016cc8956af2baa40cb67fc7d/lib/librte_mempool/rte_mempool.c#L940-L941) also calls the ``rte_mempool_set_ops_byname()`` to set the operations after calling ``rte_mempool_create_empty()``.